### PR TITLE
[BUG_FIX] Fixes #6608: CHECK(data != nullptr) causes type checking to fail

### LINKED ISSUE #6608: CHECK(data != nullptr) causes type checking to fail
Many type relations check that types are not equal to nullptr. This results in early failure of the type checker. As an example, when trying to import efficientnet-b0 (tflite-int8-model), this error occurs: 

`%0 = qnn.requantize(%images, 0.012566f, 131, 0.012566f, 3, out_dtype="int8") an internal invariant was violated while typechecking your program [17:21:40] /home/vega/github/incubator-tvm/src/relay/qnn/op/requantize.cc:261: Check failed: data != nullptr`

Instead of applying a check, the type relation should return false if data is equal to nullptr. This way, the type checker can continue to try to figure out what the type of data is. If the type checker really cannot find the type, the type checker itself should throw an error. 

cc @jwfromm @areusch @vegaluisjose 

I should be able to get a fix up soon.

### PR DESCRIPTION
**Pull request by electriclilies**
Some type rels check that argument types are not null. In certain cases, this can cause the type inferencer to exit prematurely, and the model cannot be run.

Instead of checking that argument types are not null, type rels should just return false, and allow the type checker to determine if the type of arguments cannot be inferred. 

I believe this PR fixes all of the type rels with this issue, however I could have missed some. If you find another, please let me know. 

See issue #6608 for more details.

cc @jwfromm @vegaluisjose @areusch @mbrookhart

### CODE REVIEW

File: src/relay/op/tensor/transform.cc [hunk 1]
```diff
@@ -1979,7 +1985,11 @@ bool StridedSliceRel(const Array<Type>& types, int num_inputs, const Attrs& attr
   const StridedSliceAttrs* param = attrs.as<StridedSliceAttrs>();
   CHECK(param != nullptr);
   const auto* data = types[0].as<TensorTypeNode>();
-  CHECK(data != nullptr);
+
+  if (data != nullptr) {
```
Comment (mbrookhart) [inline src/relay/op/tensor/transform.cc:1989]: if (data == nullptr)

**Review by mbrookhart — CHANGES_REQUESTED** [review]
One logical bug, but otherwise looks great :100:

File: src/relay/op/tensor/transform.cc (hunk 1 above)
Comment (electriclilies) [reply src/relay/op/tensor/transform.cc:1989]: whoopsies

Comment (electriclilies) [reply src/relay/op/tensor/transform.cc]: Should be fixed now.

**Review by mbrookhart — APPROVED** [review]

**Review by vegaluisjose — APPROVED** [review]

**Comment by tqchen** [comment]
Thanks @electriclilies @vegaluisjose @mbrookhart !

**Pull request merged**